cmd/scollector/collectors/aws.go

```diff
@@ -0,0 +1,98 @@
+package collectors
+
+import (
+	"fmt"
+	"time"
+
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+	"bosun.org/slog"
+
+	"github.com/awslabs/aws-sdk-go/aws"
+	"github.com/awslabs/aws-sdk-go/gen/ec2"
+	//"github.com/awslabs/aws-sdk-go/gen/elb"
+	"github.com/awslabs/aws-sdk-go/gen/cloudwatch"
+)
+
+const (
+	awsCPU string = "aws.ec2.cpu"
+)
+
+func AWS(accessKey, secretKey, region string) {
+	slog.Info("warming up the AWS system")
+	collectors = append(collectors, &IntervalCollector{
+		F: func() (opentsdb.MultiDataPoint, error) {
+			return c_aws(accessKey, secretKey, region)
+		},
+		name: fmt.Sprintf("aws-%s", region),
+	})
+}
+
+func c_aws(accessKey, secretKey, region string) (opentsdb.MultiDataPoint, error) {
+	var md opentsdb.MultiDataPoint
+
+	slog.Info("c_aws called")
+
+	creds := aws.Creds(accessKey, secretKey, "")
+	if creds == nil {
+		slog.Warning("Unable to make creds")
+	}
+	ecc := ec2.New(creds, region, nil)
+	if ecc == nil {
+		slog.Warning("Unable to login to EC2")
+	}
+	//elb := elb.New(creds, region, nil)
+
+	cw := cloudwatch.New(creds, region, nil)
+	if cw == nil {
+		slog.Warning("Unable to login to CloudWatch")
+	}
+	instances, err := AWSGetInstances(*ecc)
+	if err != false {
+		// Do something useful in error
+	}
+	for _, instance := range instances {
+		AWSGetCPU(*cw, &md, instance)
+	}
+	return md, nil
+
+}
+
+func AWSGetInstances(ecc ec2.EC2) ([]ec2.Instance, bool) {
```

Comment:
So should be ([]ec2.Instance, error) for the return types. If you need to create you own error you can use fmt.Errorf() - see  http://blog.golang.org/error-handling-and-go